mypy/newsemanal/semanal.py

```diff
@@ -4149,6 +4144,9 @@ def defer(self) -> None:
 
         This must be called if something in the current target is
         incomplete or has a placeholder node.
+
+        This must not be called during the final analysis iteration!
+        Instead, an error should be generated.
```

Comment:
As a follow-up to this, it might be worth adding an assert to this effect and seeing if anything more shakes out.

Reply:
A good idea! We already have a check for this later, but generating an error here would produce a useful traceback.